xoinvader/tests/test_collision.py

```diff
@@ -1,10 +1,147 @@
-import unittest
+import pytest
+import xoinvader.collision as collision
+from xoinvader.collision import Collider, CollisionManager, TypePair
+from xoinvader.utils import Point
 
-import xoinvader.collision
 
+def test_collision():
+    with pytest.raises(ValueError):
+        c = Collider("", [], None)
+    cm = CollisionManager()
+    c1 = Collider(
+        "",
+        [
+            "##",
+            "##",
+        ],
+        Point(0, 0))
 
-class TestCollision(unittest.TestCase):
-    unittest.skip("Stub.")
+    c2 = Collider(
+        "",
+        [
+            "..",
+            ".."
+        ],
+        Point(0, 0))
 
-    def test_collision(self):
-        pass
+    assert cm.check_collision(c1, c2) is None
+
+    c3 = Collider(
+        "",
+        [
+            "##",
+            "##"
+        ],
+        Point(0, 0))
+    assert cm.check_collision(c1, c3) is not None
+
+    c3._pos = Point(1, 0)
+    assert cm.check_collision(c1, c3) is not None
+    c3._pos = Point(0, 1)
+    assert cm.check_collision(c1, c3) is not None
+    c3._pos = Point(1, 1)
+    assert cm.check_collision(c1, c3) is not None
+
+    c3._pos = Point(0, 2)
+    assert cm.check_collision(c1, c3) is None
+    c3._pos = Point(2, 0)
+    assert cm.check_collision(c1, c3) is None
+
+    c4 = Collider(
+        "",
+        [
+            "..",
+            ".#",
+        ],
+        Point(0, 0))
+    collision.DEBUG = True
+    assert cm.check_collision(c1, c4) is not None
+    c4._pos = Point(1, 0)
+    assert cm.check_collision(c1, c4) is None
+
+
+def test_type_pair():
+    p1 = TypePair("t1", "t2")
+    p2 = TypePair("t2", "t1")
+    assert p1._ab == p2._ab
+    assert p1 == p2
+    assert hash(p1) == hash(p2)
+
+
+def test_collision_manager():
+    import gc
+    gc.collect()
+    assert Collider.__manager__ is None
+
+    class MockManager(object):
```

Comment:
```python
    m = MockManager()
    Collider.__manager__ = m
    c = Collider("", [], None)
    mm = CollisionManager()
    with pytest.raises(ValueError):
        mm._add_collision(c, "other", lambda: _)
```
Last line shoud rise when collider `c` is not present in current CollisionManager.